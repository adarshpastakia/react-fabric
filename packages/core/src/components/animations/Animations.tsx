/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import classNames from "classnames";
import { useMemo } from "react";
import { useMemoDebugger } from "../../hooks/useEffectDebugger";
import {
  type AriaProps,
  type ColorType,
  type CssProp,
  type PaletteType,
} from "../../types";
import { getColor, getColorClass } from "../../utils";
import classes from "./Animations.module.css";

export interface AnimationProps extends CssProp, AriaProps {
  type?: "info" | "check" | "cross" | "question" | "exclaim";
  /**
   * icon color
   */
  color?: ColorType | PaletteType;
  /**
   * icon mark color
   */
  strokeColor?: ColorType | PaletteType;
}

export const AnimationIndicator = ({
  className,
  color,
  strokeColor,
  type,
  ...aria
}: AnimationProps) => {
  const styles = useMemoDebugger(
    () => {
      const s: KeyValue = {};
      if (color) {
        s.color = getColor(color);
      }
      if (strokeColor) {
        s.stroke = getColor(strokeColor);
      }
      return s;
    },
    [color, strokeColor],
    "Animation styles",
  );
  const path = useMemo(() => {
    if (type === "check") return "M14.1 27.2l7.1 7.2 16.7-16.8";
    if (type === "cross") return "M14.1 14.1l23.8 23.8 m0,-23.8 l-23.8,23.8";
    if (type === "question")
      return "M18.949 17.621c0.426-0.787 1.354-1.6 2.108-2.223 1.279-0.984 3.87-1.778 5.805-1.778 3.837 0 6.189 2.689 6.189 6.034 0 2.23-1.193 4.214-3.751 6.346-2.886 2.361-3.497 3.362-3.497 5.788v0c0-2.263 0.689-3.509 3.476-5.772 2.689-2.23 3.772-3.968 3.772-6.362 0-3.772-2.492-6.034-6.789-6.034-2.132 0-3.903 0.722-5.28 1.804-0.82 0.623-1.476 1.377-2.033 2.197zM26 41.329v0-2.948z";
    if (type === "exclaim")
      return "M26.611 10.873c-0.002 1.015-0.002 2.071-0.002 3.086v11.777c0 3.086 0.002 2.232 0 3.288h-0.018c-0.001-1.056 0-0.202 0-3.288v-11.777c0-1.015 0-2.071-0.002-3.086zM26.59 37.472h0.018v3.655h-0.018z";
    return "M26 22L26 41M27 22L27 41M25 22L25 41M26 11L26 16M27 11L27 16M25 11L25 16Z";
  }, [type]);
  /** ***************** component *******************/
  return (
    <svg
      data-ref="animated-indicator"
      className={classNames(
        classes.checkAnimation,
        className,
        "inline-block rounded-full",
        color && getColorClass(color),
      )}
      style={styles}
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 52 52"
      aria-hidden="true"
      {...aria}
    >
      <circle className={classes.circlePart} cx="26" cy="26" r="25" />
      <path
        className={classNames(
          classes.iconPart,
          getColorClass(strokeColor ?? "invert"),
        )}
        d={path}
      />
    </svg>
  );
};

export const AnimationBars = ({
  color = "tint-700",
  size = "2rem",
}: {
  color?: ColorType | string;
  size?: number | string;
}) => {
  return (
    <div
      className={classNames(
        "absolute inset-0 z-10 flex justify-center items-center cursor-wait",
      )}
    >
      <div
        className={classNames(
          "outline outline-tint-500/20 p-4 bg-base/50 backdrop-blur-sm rounded",
        )}
      >
        <div
          className={classNames(classes.bars, getColorClass(color + "-500/50"))}
          style={{ width: size }}
        />
      </div>
    </div>
  );
};

export const AnimationSpinner = ({
  color = "tint-700",
  size = "2rem",
}: {
  color?: ColorType | string;
  size?: number | string;
}) => {
  return (
    <div
      className={classNames(
        "absolute inset-0 z-10 flex justify-center items-center cursor-wait",
      )}
    >
      <div
        className={classNames(
          "outline outline-tint-500/20 p-4 bg-base/50 backdrop-blur-sm rounded",
        )}
      >
        <div
          className={classNames(classes.spinner, getColorClass(color))}
          style={{ width: size }}
        />
      </div>
    </div>
  );
};

export const Loading = () => {
  return (
    <div
      className={classNames(
        classes.loader,
        "area-[loader] relative overflow-x-clip",
      )}
    >
      <div className="absolute z-10 py-px inset-x-0 top-0 after:h-0.5 after:block after:bg-primary-500" />
    </div>
  );
};

export const Skeleton = () => {
  return (
    <div className={classNames("p-2 animate-pulse flex gap-2 max-w-lg")}>
      <div className="bg-tint-100 rounded aspect-square h-12" />
      <div className="flex-1 flex flex-col gap-2">
        <div className="bg-tint-100 h-3 rounded-full" />
        <div className="flex gap-2">
          <div className="bg-tint-100 h-2 rounded-full basis-[30%]" />
          <div className="bg-tint-100 h-2 rounded-full basis-[60%]" />
        </div>
        <div className="flex gap-2">
          <div className="bg-tint-100 h-2 rounded-full basis-[30%]" />
          <div className="bg-tint-100 h-2 rounded-full basis-[60%]" />
        </div>
      </div>
    </div>
  );
};
