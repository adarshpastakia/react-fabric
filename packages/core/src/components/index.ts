/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

export { ActionLabel } from "./actionLabel/ActionLabel";
export {
  AnimationBars,
  AnimationIndicator,
  AnimationSpinner,
  Loading,
  Skeleton,
} from "./animations/Animations";
export { Avatar } from "./avatar/Avatar";
export { AvatarGroup } from "./avatar/AvatarGroup";
export { Badge, getBadgeProps } from "./badge/Badge";
export { Breadcrumbs } from "./breadcrumbs/Breadcrumbs";
export { Button } from "./button/Button";
export { ButtonGroup } from "./button/ButtonGroup";
export { ConfirmButton } from "./button/ConfirmButton";
export { ToggleButtonGroup } from "./button/ToggleButtonGroup";
export { Callout } from "./callout/Callout";
export { Card } from "./card/Card";
export { CardCover } from "./card/CardCover";
export { FlipContent, type FlipContentRef } from "./card/FlipContent";
export { Chip } from "./chip/Chip";
export { Collapsable } from "./collapsable/Collapsable";
export { Divider } from "./divider/Divider";
export { Dropdown, DropdownDismiss } from "./dropdown/Dropdown";
export { DropdownTool } from "./dropdown/DropdownTool";
export { EmptyContent } from "./empty/EmptyContent";
export { Icon } from "./icon/Icon";
export { Image, Video } from "./media/Media";
export { ContextMenu } from "./menu/ContextMenu";
export { Menu } from "./menu/Menu";
export { MenuItem } from "./menu/MenuItem";
export { Meter } from "./meter/Meter";
export { Navigator } from "./navigator/Navigator";
export { Panel } from "./panel/Panel";
export { PanelGroup } from "./panel/PanelGroup";
export { PanelStack } from "./panel/PanelStack";
export { ProgressBar } from "./progress/ProgressBar";
export { ProgressCircle } from "./progress/ProgressCircle";
export { Tab } from "./tabpanel/Tab";
export { TabPanel } from "./tabpanel/TabPanel";
