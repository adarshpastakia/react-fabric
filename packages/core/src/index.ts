/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

export * from "./components";
export * from "./core";
export { addTranslationBundle } from "./i18n";
export { Flyout, Modal, Tooltip } from "./overlays";
export * from "./typography";

export * from "./hooks/useClientService";
export * from "./hooks/useDebounce";
export * from "./hooks/useEffectDebugger";
export * from "./hooks/useIsDark";
export * from "./hooks/useIsRtl";
export * from "./hooks/useNavigator";
export * from "./hooks/useOverlayService";
export * from "./hooks/useResize";
export * from "./hooks/useResizeObserver";
export * from "./hooks/useStorage";

export { HotKey } from "./hotkeys/HotKey";
export { HotKeyLabel } from "./hotkeys/HotKeyLabel";

export { CoreIcons } from "./types/icons";

export {
  ApplicationProvider,
  useApplicationContext,
  useGlobals,
  useNotificationService,
} from "./context/Global";
