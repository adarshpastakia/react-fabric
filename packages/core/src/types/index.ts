/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
/* istanbul ignore file */

import { type Placement } from "@floating-ui/react";
import { type JSXElementConstructor, type Ref } from "react";

export type SizeType = "xs" | "sm" | "md" | "lg" | "xl";

export type SizeTypeWithZero = "xs" | "sm" | "md" | "lg" | "xl" | 0;

export type ColorType =
  | "primary"
  | "accent"
  | "info"
  | "danger"
  | "success"
  | "warning";

export type PaletteType =
  | "scarlet"
  | "pumpkin"
  | "marigold"
  | "avacado"
  | "jade"
  | "denim"
  | "iris"
  | "lilac"
  | "coral"
  | "wood"
  | "silver"
  | "steel"
  | "olive"
  | "sand";

export type ColorWithGrays = ColorType | "invert" | "dimmed" | "muted";

export const ThemeColors = [
  "primary",
  "accent",
  "info",
  "danger",
  "success",
  "warning",
  "invert",
  "dimmed",
  "muted",
  "gray",
  "dark",
  "light",
];

export interface TooltipType {
  /**
   * tooltip content
   */
  content?: string | JSX.Element;
  /**
   * tooltip color
   */
  color?: ColorType;
  /**
   * tooltip placement
   */
  placement?: Placement;
}

export interface BadgeType {
  /**
   * badge value
   */
  value?: number | string;
  /**
   * max value for number display
   */
  max?: number;
  /**
   * ping animation
   */
  ping?: boolean;
  /**
   * badge CSS className
   */
  className?: string;
}

export interface CssProp {
  /**
   * css classname(s)
   */
  className?: string;
}

export interface TestProps {
  /**
   * data attribute for test id
   */
  "data-testid"?: string;
  /**
   * data attribute for test value
   */
  "data-test-value"?: AnyObject;
}

export type Elements<T> =
  | React.ReactElement<T>
  | React.ReactPortal
  | React.ReactNode
  | string
  | number
  | boolean
  | null
  | undefined;

export type AriaProps = React.AriaAttributes;

/**
 * For wrapping components
 */

export interface RefProp<T = HTMLElement> {
  ref?: Ref<T>;
}

/* *********** */

// eslint-disable-next-line @typescript-eslint/no-invalid-void-type
export type CallbackReturn<T = unknown> = void | T | Promise<T | void>;

export type ReactTag = keyof JSX.IntrinsicElements | JSXElementConstructor<any>;

export type PropsOf<AsTag extends ReactTag> = AsTag extends React.ElementType
  ? Omit<React.ComponentProps<AsTag>, "ref">
  : never;

export type PolymorphicProps<AsTag extends ReactTag> = Omit<
  PropsOf<AsTag>,
  "children" | "as" | "className"
> & {
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  as?: AsTag;
};

export interface IconProps {
  /**
   * svg path / image url
   */
  icon?: string;
  /**
   * flip icon in rtl
   */
  rtlFlip?: boolean;
  /**
   * icon background color
   */
  iconBg?: ColorType | string;
  /**
   * icon color
   */
  iconColor?: ColorType | string;
}

export interface CollapseProps {
  /**
   * Allow collapsing
   */
  collapsable?: boolean;
  /**
   * Default collapsed
   */
  collapsed?: boolean;
  /**
   * Collapse event
   */
  onCollapse?: (state: boolean) => void;
}
export interface ExpandProps {
  /**
   * Allow expanding
   */
  expandable?: boolean;
  /**
   * Default expanded
   */
  expanded?: boolean;
  /**
   * Expand event
   */
  onExpand?: (state: boolean) => void;
}

export interface TooltipProp {
  /**
   * floating tooltip
   */
  tooltip?: string | TooltipType;
}

export interface ChildProp<T = JSX.Element> {
  children: Elements<T>;
}

export interface ChildrenProp<T = JSX.Element> {
  children: Elements<T> | Array<Elements<T>>;
}

export interface PositionObject {
  top: number;
  left: number;
}

export interface SizeObject {
  width: number;
  height: number;
}
