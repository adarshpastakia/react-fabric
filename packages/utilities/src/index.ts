/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

/* istanbul ignore file */

export * from "./_ascii";
export * from "./_boundingBox";
export { getBoundingBox, getBox } from "./_boundingBox";
export { default as Countries } from "./_countries";
export type { Country } from "./_countries";
export * from "./_debounce";
export * from "./_dedupe";
export { _fetch as fetch } from "./_fetch";
export { default as FileUtil } from "./_fileType";
export * from "./_format";
export * from "./_getByPath";
export * from "./_hash";
export * from "./_interpolate";
export * from "./_isEqual";
export * from "./_isType";
export * from "./_mergeRefs";
export * from "./_tokenize";
export { getImageColorset } from "./getImageColorset";
export { useLogger } from "./useLogger";
export * from "./yup";

export const EMPTY_ARRAY = [];
Object.freeze(EMPTY_ARRAY);

export const EMPTY_OBJECT = {};
Object.freeze(EMPTY_OBJECT);
