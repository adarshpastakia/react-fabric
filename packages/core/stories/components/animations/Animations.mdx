import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { AnimationIndicator, Collapsable } from "../../../src";
import * as AnimationsStories from "./Animations.stories";
import * as BarsStories from "./Bars.stories";
import * as SpinnerStories from "./Spinner.stories";
import * as SkeletonStories from "./Skeleton.stories";

<Meta of={AnimationsStories} />

<Title>Animations</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Indicators</Subheading>
  </div>
</Anchor>

<Description of={AnimationIndicator} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={AnimationsStories._Animations} />
</Collapsable>

<Anchor storyId="bars">
  <Subheading>Bars</Subheading>
</Anchor>
<Description of={BarsStories._Bars} />
<Canvas withToolbar of={BarsStories._Bars} />
<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={BarsStories._Bars} />
</Collapsable>

<Anchor storyId="spinner">
  <Subheading>Spinner</Subheading>
</Anchor>
<Description of={SpinnerStories._Spinner} />
<Canvas withToolbar of={SpinnerStories._Spinner} />
<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={SpinnerStories._Spinner} />
</Collapsable>

<Anchor storyId="skeleton">
  <Subheading>Skeleton</Subheading>
</Anchor>
<Description of={SkeletonStories._Skeleton} />
<Canvas withToolbar of={SkeletonStories._Skeleton} />

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="sample">
  <Subheading>Example</Subheading>
</Anchor>

```tsx
import { AnimationCheck, AnimationBars } from "@react-fabric/core";

export const Example = () => {
  return (
    <AnimationCheck/>
    <AnimationBars/>
  );
};
```
