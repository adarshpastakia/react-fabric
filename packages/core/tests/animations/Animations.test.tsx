/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { composeStories } from "@storybook/react";
import { render } from "@testing-library/react";
import { Loading } from "../../src";
import * as stories from "../../stories/components/animations/Animations.stories";
import * as barStories from "../../stories/components/animations/Bars.stories";
import * as skeletonStories from "../../stories/components/animations/Skeleton.stories";
import * as spinnerStories from "../../stories/components/animations/Spinner.stories";

const { Tester } = composeStories(stories);
const { Tester: BarTester } = composeStories(barStories);
const { Tester: SkeletonTester } = composeStories(skeletonStories);
const { Tester: SpinnerTester } = composeStories(spinnerStories);

describe("Animations", () => {
  it("should render indicator", () => {
    const fragment = render(
      <div>
        <Tester />
        <Tester type="info" />
        <Tester type="check" />
        <Tester type="question" />
        <Tester type="cross" color="lilac" />
        <Tester type="exclaim" strokeColor="lilac" />
      </div>,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render bars", () => {
    const fragment = render(
      <div>
        <BarTester />
      </div>,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render skeleton", () => {
    const fragment = render(
      <div>
        <SkeletonTester />
      </div>,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render spinner", () => {
    const fragment = render(
      <div>
        <SpinnerTester />
      </div>,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render loader", () => {
    const fragment = render(
      <div className="relative">
        <Loading />
      </div>,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });
});
