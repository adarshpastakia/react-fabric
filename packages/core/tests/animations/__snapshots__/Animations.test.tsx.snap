// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animations should render bars 1`] = `"<div></div><div><div class="relative"><div class="absolute inset-0 z-10 flex justify-center items-center cursor-wait"><div class="outline outline-tint-500/20 p-4 bg-tint-50/50 backdrop-blur-sm rounded"><div class="fabric-bars" style="width: 2rem;"></div></div></div></div></div>"`;

exports[`Animations should render indicator 1`] = `"<div><div tabindex="0"><div style="display: contents;"><div class="flex gap-2 items-center p-4 h-12"><svg data-ref="animated-indicator" class="fabric-checkAnimation inline-block rounded-full" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 52 52" aria-hidden="true"><circle class="fabric-circlePart" cx="26" cy="26" r="25"></circle><path class="fabric-iconPart" d="M26 22L26 41M27 22L27 41M25 22L25 41M26 11L26 16M27 11L27 16M25 11L25 16Z"></path></svg></div></div><div style="display: contents;"><div class="flex gap-2 items-center p-4 h-12"><svg data-ref="animated-indicator" class="fabric-checkAnimation inline-block rounded-full" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 52 52" aria-hidden="true"><circle class="fabric-circlePart" cx="26" cy="26" r="25"></circle><path class="fabric-iconPart" d="M26 22L26 41M27 22L27 41M25 22L25 41M26 11L26 16M27 11L27 16M25 11L25 16Z"></path></svg></div></div><div style="display: contents;"><div class="flex gap-2 items-center p-4 h-12"><svg data-ref="animated-indicator" class="fabric-checkAnimation inline-block rounded-full" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 52 52" aria-hidden="true"><circle class="fabric-circlePart" cx="26" cy="26" r="25"></circle><path class="fabric-iconPart" d="M14.1 27.2l7.1 7.2 16.7-16.8"></path></svg></div></div><div style="display: contents;"><div class="flex gap-2 items-center p-4 h-12"><svg data-ref="animated-indicator" class="fabric-checkAnimation inline-block rounded-full" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 52 52" aria-hidden="true"><circle class="fabric-circlePart" cx="26" cy="26" r="25"></circle><path class="fabric-iconPart" d="M18.949 17.621c0.426-0.787 1.354-1.6 2.108-2.223 1.279-0.984 3.87-1.778 5.805-1.778 3.837 0 6.189 2.689 6.189 6.034 0 2.23-1.193 4.214-3.751 6.346-2.886 2.361-3.497 3.362-3.497 5.788v0c0-2.263 0.689-3.509 3.476-5.772 2.689-2.23 3.772-3.968 3.772-6.362 0-3.772-2.492-6.034-6.789-6.034-2.132 0-3.903 0.722-5.28 1.804-0.82 0.623-1.476 1.377-2.033 2.197zM26 41.329v0-2.948z"></path></svg></div></div><div style="display: contents;"><div class="flex gap-2 items-center p-4 h-12"><svg data-ref="animated-indicator" class="fabric-checkAnimation inline-block rounded-full" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 52 52" aria-hidden="true"><circle class="fabric-circlePart" cx="26" cy="26" r="25"></circle><path class="fabric-iconPart" d="M14.1 14.1l23.8 23.8 m0,-23.8 l-23.8,23.8"></path></svg></div></div><div style="display: contents;"><div class="flex gap-2 items-center p-4 h-12"><svg data-ref="animated-indicator" class="fabric-checkAnimation inline-block rounded-full" style="stroke: var(--color-lilac);" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 52 52" aria-hidden="true"><circle class="fabric-circlePart" cx="26" cy="26" r="25"></circle><path class="fabric-iconPart" d="M26.611 10.873c-0.002 1.015-0.002 2.071-0.002 3.086v11.777c0 3.086 0.002 2.232 0 3.288h-0.018c-0.001-1.056 0-0.202 0-3.288v-11.777c0-1.015 0-2.071-0.002-3.086zM26.59 37.472h0.018v3.655h-0.018z"></path></svg></div></div></div></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div><div style="z-index: var(--z-overlay);" class="fabric-container__overlays pointer-events-none fixed inset-0 p-4 bg-transparent"></div><div style="z-index: var(--z-notifications);" class="fabric-container__notifs flex flex-nowrap pointer-events-none fixed inset-0 p-2 bg-transparent flex-col-reverse items-center justify-end"></div><div style="z-index: var(--z-notifications);" class="fabric-container__toasts flex flex-nowrap pointer-events-none fixed inset-0 p-4 bg-transparent flex-col items-end justify-end overflow-auto scroll-hide"></div>"`;

exports[`Animations should render loader 1`] = `"<div></div><div></div><div></div><div></div><div><div class="relative"><div class="fabric-loader area-[loader] relative overflow-x-clip" data-ref="loading"><div class="absolute z-10 py-px inset-x-0 top-0 after:h-0.5 after:block after:bg-primary-500"></div></div></div></div>"`;

exports[`Animations should render skeleton 1`] = `"<div></div><div></div><div><div class="relative"><div class="p-2 animate-pulse flex gap-2 max-w-lg" data-ref="skeleton"><div class="bg-tint-100 rounded aspect-square h-12"></div><div class="flex-1 flex flex-col gap-2"><div class="bg-tint-100 h-3 rounded-full"></div><div class="flex gap-2"><div class="bg-tint-100 h-2 rounded-full basis-[30%]"></div><div class="bg-tint-100 h-2 rounded-full basis-[60%]"></div></div><div class="flex gap-2"><div class="bg-tint-100 h-2 rounded-full basis-[30%]"></div><div class="bg-tint-100 h-2 rounded-full basis-[60%]"></div></div></div></div></div></div>"`;

exports[`Animations should render spinner 1`] = `"<div></div><div></div><div></div><div><div class="relative"><div class="absolute inset-0 z-10 flex justify-center items-center cursor-wait" data-ref="spinner"><div class="outline outline-tint-500/20 p-4 bg-tint-50/50 backdrop-blur-sm rounded"><div class="fabric-spinner" style="width: 2rem;"></div></div></div></div></div>"`;
